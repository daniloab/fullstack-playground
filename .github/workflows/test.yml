name: Tests

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '14'
      - run: yarn
      - run: yarn jest