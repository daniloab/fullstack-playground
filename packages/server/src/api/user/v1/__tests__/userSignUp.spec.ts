import { clearDbAndRestartCounters, connectMongoose, disconnectMongoose } from '../../../../../test';

import { createApiCall } from '../../../../../test/helper';
import { createUser } from '../../../../../test/createResource';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

const url = '/api/user/v1/signup';

it('should sign up a new user', async () => {
  const payload = {
    name: 'Joana User',
    password: '123456',
    email: 'test@test.com',
  };

  const response = await createApiCall({ url, payload });

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});

it('should not sign up a user already registered', async () => {
  await createUser({ email: 'test@test.com' });

  const payload = {
    name: 'Joana User',
    password: '123456',
    email: 'test@test.com',
  };

  const response = await createApiCall({ url, payload });

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(400);
});
