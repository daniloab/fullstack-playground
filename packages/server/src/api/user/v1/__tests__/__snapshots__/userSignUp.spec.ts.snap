// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not sign up a user already registered 1`] = `
Object {
  "message": "User already registered",
  "status": "ERROR",
  "user": null,
}
`;

exports[`should sign up a new user 1`] = `
Object {
  "message": "User successfully created",
}
`;
